pennylane/measurements/shots.py

```diff
@@ -185,6 +186,14 @@ def __repr__(self):
         """The representation of the class"""
         return f"Shots(total_shots={self.total_shots}, shot_vector={self.shot_vector})"
 
+    def __eq__(self, other):
+        """Equality between Shot instances."""
+        return self.total_shots == other.total_shots and self.shot_vector == other.shot_vector
+
+    def __hash__(self):
+        """Hash for a given Shot instance."""
+        return hash((self.total_shots, self.shot_vector))  # hash the tuple of these two attributes
```

Comment:
the shot_vector itself should be sufficient, no? I'm good to use `total_shots` in the `__eq__` computation because it's gotta be faster if the total_shots are not equal 😄

Reply:
Double checked, it should indeed be sufficient.

Reply:
Nice call! Added 👍🏼